lcd/emonPiLCD.py

```diff
@@ -344,8 +345,8 @@ def __init__(self, logger):
         # Scan I2C bus for LCD I2C addresses as defined in led_i2c, we have a couple of models of LCD which have different addreses that are shipped with emonPi. First I2C device to match address is used.
         self.logger = logger
         for i2c_address in lcd_i2c:
-            lcd_status = subprocess.check_output([path+"/emonPiLCD_detect.sh", "%s" % i2c_address])
-            if lcd_status.rstrip() == 'True':
+            lcd_status=subprocess.run([path+"/emonPiLCD_detect.sh", "%s" % i2c_address])
```

Comment:
I'm not sure what's changing here or why. `run` is returning a `CompletedProcess` instance which is always True, which will break this logic, always choosing the first address.

Reply:
correct : my stuff is totally wrong
going to modify in order to propose something working :-)

Reply:
I think there are more problems with this code, although I haven't looked for a while, like how the code in the initialiser can fail but it has no way of returning errors, and that bash script could easily be replaced with neat python code. I'll have a closer look at this today.

Reply:
maybe the bash script could be replaced but I think I should not have replaced `check_output` by `run`...

cf https://docs.python.org/3/library/subprocess.html?highlight=check_output#subprocess.check_output

`check_output` is equivalent to `run(..., check=True, stdout=subprocess.PIPE).stdout`

actually `check_output` returns a binary and I forgot to add a decode()